### DISCUSSION TITLE
File Downlink Destination

### DISCUSSION THREAD

**Discussion by teddylambert** [nasa/fprime#1884] [Q&A] [answered]
I'm working on a project that is using the file downlink component. I'm on fprime v3.1.1 and fprime-gds v3.1.3. I tried downlinking a simple 39-byte text file, and the deployment debug statements and GDS output both show that the command was received, the downlink started, and that it was successfully completed. However, I can't find the file anywhere on my host system or on the file downlink tab of the GDS. Where do the downlinked files normally get stored? Any ideas on why I'm getting a successful completion of the command but no file?

down.txt.log file from the GDS session:
```
Received START packet with metadata:
	Size: 39
	Source: test.txt
	Destination: down.txt
Received DATA packet writing 39 bytes to offset 0
Received END packet, finishing downlink
```

event.log from GDS session:
```
2023-02-08 13:17:32,(2(0)-1675880252:442552),cmdDisp.OpCodeDispatched,1281,EventSeverity.COMMAND,Opcode 0x700 dispatched to port 3
2023-02-08 13:17:32,(2(0)-1675880252:525197),fileDownlink.SendStarted,1800,EventSeverity.ACTIVITY_HI,Downlink of 39 bytes started from test.txt to down.txt
2023-02-08 13:17:32,(2(0)-1675880252:525784),fileDownlink.FileSent,1794,EventSeverity.ACTIVITY_HI,Sent file test.txt to file down.txt
2023-02-08 13:17:32,(2(0)-1675880252:526045),cmdDisp.OpCodeCompleted,1282,EventSeverity.COMMAND,Opcode 0x700 completed
```

**Comment by LeStarch** [comment, marked as answer]
Files are stored in directories managed by the GDS on the machine running the GDS.  These directories are, by default:

```
/tmp/fprime-uplink
/tmp/fprime-downlink
```

It seems like there is an issue displaying files in the Downlink tab.  Did you see any errors reported in the GDS?

**Reply by teddylambert** [reply]
Ah, there they are! All of the files I downlinked were located successfully in `/tmp/fprime-downlink`

I didn't see any errors reported in the GDS or any GDS logs about any issues, so not sure about the Downlink tab

**Reply by LeStarch** [reply]
https://github.com/nasa/fprime/issues/1654  there was a bug.  It is now fixed.
